mlx_lm/tokenizer_utils.py

```diff
@@ -341,7 +342,7 @@ def _is_bpe_decoder(decoder):
     return isinstance(decoder, dict) and decoder.get("type", None) == "ByteLevel"
 
 
-def load_tokenizer(model_path, tokenizer_config_extra={}, eos_token_ids=None):
+def load_tokenizer(model_path, tokenizer_config_extra={}, return_tokenizer=True, eos_token_ids=None):
```

Comment:
What's the purpose of that flag?

Reply:
MLX-VLM uses the processor so I only need the detokenizer class and not the tokenizerwrapper class.

Reply:
What about the eos token ids? Do you manage those separately? They are part of the `TokenizerWrapper` and pretty important for the stopping condition.

Reply:
Yes, I manage them separately at the processor level.

Precisely, I've created a stopping criteria class. It does the same, but also allows users to pass custom tokens or a callable.

PRs:
- https://github.com/Blaizzy/mlx-vlm/pull/310
- https://github.com/Blaizzy/mlx-vlm/pull/307